dnsseed.go

```diff
@@ -0,0 +1,190 @@
+				added := amgr.AddAddresses(msg.AddrList)
+				log.Printf("Peer %v sent %v addresses, %d new",
+					p.Addr(), len(msg.AddrList), added)
+				onaddr <- struct{}{}
+			},
+			OnVersion: func(p *peer.Peer, msg *wire.MsgVersion) {
+				log.Printf("Adding peer %v with services %v",
+					p.NA().IP.String(), p.Services())
+				// notify that version is received and Peer's subnetwork ID is updated
+				onversion <- struct{}{}
+			},
+		},
+		SubnetworkID:  &wire.SubnetworkIDSupportsAll,
+		DNSSeederPeer: true,
+	}
+
+	var wg sync.WaitGroup
+	for {
+		peers := amgr.Addresses()
+		if len(peers) == 0 && amgr.AddressCount() == 0 {
+			// Add peers discovered through DNS to the address manager.
+			connmgr.SeedFromDNS(activeNetParams, defaultRequiredServices, &wire.SubnetworkIDSupportsAll, hostLookup, func(addrs []*wire.NetAddress) {
+				amgr.AddAddresses(addrs)
+			})
+			peers = amgr.Addresses()
+		}
+		if len(peers) == 0 {
+			log.Printf("No stale addresses -- sleeping for %v",
+				defaultAddressTimeout)
+			time.Sleep(defaultAddressTimeout)
+			continue
+		}
+
+		wg.Add(len(peers))
+
+		for _, addr := range peers {
+			go func(addr *wire.NetAddress) {
+				defer wg.Done()
+
+				host := net.JoinHostPort(addr.IP.String(), fmt.Sprintf("%d", int(addr.Port)))
+				p, err := peer.NewOutboundPeer(&config, host)
+				if err != nil {
+					log.Printf("NewOutboundPeer on %v: %v",
+						host, err)
+					return
+				}
+				amgr.Attempt(addr.IP)
+				conn, err := net.DialTimeout("tcp", p.Addr(),
+					defaultNodeTimeout)
+				if err != nil {
+					log.Printf("%v", err)
+					return
+				}
+				p.AssociateConnection(conn)
+
+				// Wait version messsage or timeout in case of failure.
+				select {
+				case <-onversion:
+					// Mark this peer as a good node.
+					amgr.Good(p.NA().IP, p.Services(), p.SubnetworkID())
+					// Ask peer for some addresses.
+					p.QueueMessage(wire.NewMsgGetAddr(nil), nil)
+				case <-time.After(defaultNodeTimeout):
+					log.Printf("version timeout on peer %v",
+						p.Addr())
+					p.Disconnect()
+					return
+				}
+
+				select {
+				case <-onaddr:
+				case <-time.After(defaultNodeTimeout):
+					log.Printf("getaddr timeout on peer %v",
+						p.Addr())
+					p.Disconnect()
+					return
+				}
+				p.Disconnect()
+			}(addr)
+		}
+		wg.Wait()
+		log.Printf("Sleeping for %v", defaultAddressTimeout)
+		time.Sleep(defaultAddressTimeout)
+	}
+}
+
+func main() {
+	cfg, err := loadConfig()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "loadConfig: %v\n", err)
+		os.Exit(1)
+	}
+	amgr, err = NewManager(defaultHomeDir)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "NewManager: %v\n", err)
+		os.Exit(1)
+	}
+
+	peersDefaultPort, _ = strconv.Atoi(activeNetParams.DefaultPort)
+
+	if len(cfg.Seeder) != 0 {
+		ip := net.ParseIP(cfg.Seeder)
+		if ip == nil {
+			hostAdrs, err := net.LookupHost(cfg.Seeder)
+			if err != nil {
+				log.Printf("Failed to resolve seed host: %v, %v, ignoring", cfg.Seeder, err)
```

Comment:
Can we separate errors to stderr in the entirety of this package?

If that's very complicated - forget about it (at least for now), but we should strive to use the output streams correctly where possible.